api/v1beta1/openstackdataplanenodeset_types.go

```diff
@@ -181,16 +185,58 @@ func (instance OpenStackDataPlaneNodeSet) GetAnsibleEESpec() AnsibleEESpec {
 	}
 }
 
-// DataplaneAnsibleImageDefaults default images for dataplane services
-type DataplaneAnsibleImageDefaults struct {
-	Frr                        string
-	IscsiD                     string
-	Logrotate                  string
-	NeutronMetadataAgent       string
-	NovaCompute                string
-	OvnControllerAgent         string
-	OvnBgpAgent                string
-	TelemetryCeilometerCompute string
-	TelemetryCeilometerIpmi    string
-	TelemetryNodeExporter      string
+// ContainerImages default images for dataplane services
+type ContainerImages struct {
+	Frr                        string `json:"frr,omitempty"`
+	IscsiD                     string `json:"iscsiD,omitempty"`
+	Logrotate                  string `json:"logrotate,omitempty"`
+	NeutronMetadataAgent       string `json:"neutronMetadataAgent,omitempty"`
+	NovaCompute                string `json:"novaCompute,omitempty"`
+	OvnControllerAgent         string `json:"ovnControllerAgent,omitempty"`
+	OvnBgpAgent                string `json:"ovnBgpAgent,omitempty"`
+	TelemetryCeilometerCompute string `json:"telemetryCeilometerCompute,omitempty"`
+	TelemetryCeilometerIpmi    string `json:"telemetryCeilometerIpmi,omitempty"`
+	TelemetryNodeExporter      string `json:"telemetryNodeExporter,omitempty"`
+}
```

Comment:
using a stuct instead of a hashmap of sting to sting will mean that this upgrade mechanium cannot be used for thrid party vendor integrations. for example juniper vrouter or any ohter integration that would require addtional containers.

Reply:
wouldn't juniper vrouter deployments get deployed with their own Ansible playbooks? And any env variables there can be set in the Ansible inventory or whatever. The interface we are building here can accomidate custom backends where needed (storage backends). But on the dataplane it unless it is our code that is orchestrating the deployment (i.e. our Ansible) I'm not sure we need to have a parameter to represent anything yet.